ext/mbstring/mbstring.c

```diff
@@ -2941,6 +2941,142 @@ PHP_FUNCTION(mb_strtolower)
 	RETURN_STR(mbstring_convert_case(PHP_UNICODE_CASE_LOWER, ZSTR_VAL(str), ZSTR_LEN(str), enc));
 }
 
+typedef enum {
+	MB_LTRIM = 1,
+	MB_RTRIM = 2,
+	MB_BOTH_TRIM = 3
+} MB_TRIM_MODE;
+
+static bool is_trim_wchar(uint32_t w, uint32_t *characters, size_t characters_len)
+{
+	for (size_t i = 0; i < characters_len; i++) {
+		if (w == characters[i]) {
+			return true;
+		}
+	}
+	return false;
+}
+
+static zend_string* trim_each_wchar(zend_string *str, zend_string *what, MB_TRIM_MODE mode, const mbfl_encoding *enc)
+{
+	unsigned char *in = (unsigned char*)ZSTR_VAL(str);
+	unsigned char *what_in = (unsigned char*)ZSTR_VAL(what);
+	uint32_t wchar_buf[128];
+	uint32_t what_wchar_buf[128];
+	size_t in_len = ZSTR_LEN(str);
+	size_t out_len = 0;
+	size_t what_out_len = 0;
+	size_t what_len = ZSTR_LEN(what);
+	unsigned int state = 0;
+	long left = 0;
```

Comment:
Almost forgot about this: you should use size_t for left and right. On 64-bit windows long has a smaller size than size_t: 4 bytes vs 8. That would cause issues. Best to use size_t.

Reply:
Thank you. Fixed in a2fedccf28